src/app/components/filter-group/vtuber-filter/vtuber-filter.ts

```diff
@@ -1,68 +1,93 @@
 import {
-  ConnectedPosition,
-  Overlay,
-  OverlayModule,
-} from "@angular/cdk/overlay";
-import { NgOptimizedImage } from "@angular/common";
-import { Component, ViewEncapsulation, inject, output } from "@angular/core";
+  Component,
+  HostListener,
+  ViewChild,
+  inject,
+  output,
+} from "@angular/core";
 import { MatListModule, MatSelectionListChange } from "@angular/material/list";
 import { MatTooltipModule } from "@angular/material/tooltip";
 import { AvatarPipe, NamePipe } from "src/app/shared";
 import { VTuberService } from "src/app/shared/config/vtuber.service";
-
 import { MatButtonModule } from "@angular/material/button";
 import { MatIconModule } from "@angular/material/icon";
-import animations from "../_animations";
+import { MatFormFieldModule } from "@angular/material/form-field";
+import { MatInputModule } from "@angular/material/input";
+import {
+  MatMenuModule,
+  MatMenuTrigger,
+} from "@angular/material/menu";
 
 @Component({
   standalone: true,
   imports: [
     MatTooltipModule,
     MatListModule,
-    OverlayModule,
+    MatFormFieldModule,
+    MatInputModule,
+    MatMenuModule,
     NamePipe,
     AvatarPipe,
-    NgOptimizedImage,
     MatButtonModule,
     MatIconModule,
   ],
   selector: "vts-vtuber-filter",
   templateUrl: "vtuber-filter.html",
-  animations,
-  encapsulation: ViewEncapsulation.None,
 })
 export class VTuberFilter {
-  closeScrollStrategy = inject(Overlay).scrollStrategies.close();
-  positions: ConnectedPosition[] = [
-    {
-      originX: "start",
-      originY: "bottom",
-      overlayX: "start",
-      overlayY: "top",
-      offsetY: 10,
-    },
-    {
-      originX: "start",
-      originY: "top",
-      overlayX: "start",
-      overlayY: "bottom",
-      offsetY: -10,
-    },
-  ];
-
   private vtubers = inject(VTuberService);
 
   selected: Set<string> = new Set();
+  _vtubers = this.vtubers.selected().map((v) => v.vtuberId);
 
-  _isOpen = false;
+  selectedChange = output<Set<string>>();
 
-  _vtubers = this.vtubers.selected();
+  searchText = "";
 
-  selectedChange = output<Set<string>>();
+  @ViewChild("filterMenuTrigger") filterMenuTrigger: MatMenuTrigger | undefined;
+  @HostListener("window:scroll", [])
+  // Handles scroll on close for mat-menu
+  // Still wondering if there's a solution invloving scroll strategy, but this is the best I can do
+  scrollHandler() {
+    if (this.filterMenuTrigger) {
+      this.closeMenu();
+    }
+  }
+
+  handleOnSearchChange(e: EventTarget) {
+    this.searchText = (e as HTMLInputElement).value.toLowerCase();
+
+    this.updateSearchQuery(this.searchText);
```

Comment:
don't forget to `trim` the input string, searching for whitespace doesn't make any sense